### DISCUSSION TITLE
BLE Distance Wildly unstable

### DISCUSSION THREAD

**Discussion by AshtonKem** [mKeRix/room-assistant#526] [Q&A]
I've setup room-assistant with BLE & the companion app on my spare RPI 3. With both the phone and the the RPI in the room sitting still, I've seen a variance of well over 2m in the estimated distance. Is this an expected outcome? Because it sure seems like it won't be possible to do significant triangulation with such a wild variance.

**Comment by mKeRix** [comment]
It's somewhat expected, it really depends on your environment though. Bluetooth isn't an exact form of tracking, especially when there's walls, other radio waves and longer distances. It's also why I've been a bit skeptical of the triangulation approach (even though others have proven me wrong there, someone posted a whole guide over in the Home Assistant forums).

room-assistant already smoothes the BLE data a bit to get better distance estimates, but I took away the configuration options for that when I wrote v2.x (because nobody really used it in v1.x). I suppose further tuning of those parameters could help reduce this variance. In addition, I have hopes that #507 will be an even better approach than triangulation and what is integrated into room-assistant right now.
